### PR TITLE
ENTST-380: add checkbox include highlights to x dash gift article modal

### PR DESCRIPTION
**Pull request by abshirahmed**
## Description
This PR introduces two features
- Highlights checkbox that creates a share with highlights included
- A new message is introduced to B2B recipients in the x-gift-article prompting the users to save their highlights in order to re-share the article with the received highlights.

Changes:

- Created a new message `HighlightRecipientMessage` which will be shown to B2B users only if they open a shared article with highlights( highlights query parameter should be in the url)

- Created a new success message `HighlightSuccessMessage` which will appear after the user clicks save the highlights.

- Created a `HighlightCheckbox` component, which will appear after the user has saved the shared highlights or when the user has created highlights on their own.

- Created `HighlightsSection` component to be a wrapper of the highlights checkbox, the recipient message for saving highlights and the success message afterwards.

- Created a separate `highlightsApi` client to be used only for requests to the user-annotations-api
- Updated the `entrepriseApi`

## Jira Tickets 

Highlights checkbox ticket
https://financialtimes.atlassian.net/browse/ENTST-352

Recipient message ticket
https://financialtimes.atlassian.net/browse/ENTST-397

### Sharing article with existing highlights
https://github.com/Financial-Times/x-dash/assets/22678655/b17b8ca1-494c-4ab1-b18a-b7fb4353cbcc

### Sharing article with no highlights
Screenshot 2023-05-04 at 15 28 14

### Sharing article with received highlights

https://github.com/Financial-Times/x-dash/assets/22678655/539e4301-7fb2-4487-9a01-50d71b142882

### CODE REVIEW

File: components/x-gift-article/src/HighlightSection.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	background-color: inherit !important;
+}
+
+.x-gift-article__highlight-shared {
+	@include oTypographySans($scale: -1);
+	background-color: oColorsMix(black, paper, 5);
+	width: fit-content;
+	padding: 4px 8px;
+	text-transform: uppercase;
+	color: #333333;
```
Comment (fabiosikansi) [inline components/x-gift-article/src/HighlightSection.scss:27]: colour should defined from oColors

File: components/x-gift-article/src/HighlightSection.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+.x-gift-article__checkbox-span::before {
+	background-color: inherit !important;
+}
+
+.x-gift-article__highlight-shared {
+	@include oTypographySans($scale: -1);
+	background-color: oColorsMix(black, paper, 5);
+	width: fit-content;
+	padding: 4px 8px;
```
Comment (fabiosikansi) [inline components/x-gift-article/src/HighlightSection.scss:25]: spacing should me defined from oSpacing

File: components/x-gift-article/src/HighlightSection.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
+@import '@financial-times/o-typography/main';
+@import '@financial-times/o-colors/main';
+
+.x-gift-article__checkbox {
+	border-top: 1px solid oColorsMix(black, paper, 20);
+	padding-top: 22px;
+	margin-bottom: 12px;
```
Comment (fabiosikansi) [inline components/x-gift-article/src/HighlightSection.scss:7]: spacing should be defined from oSpacing standards

File: components/x-gift-article/src/HighlightSection.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	hasHighlights,
+	includeHighlightsHandler,
+	includeHighlights,
+	isGiftUrlCreated
+}) => {
+	if (shareType === ShareType.enterprise && hasHighlights) {
+		if (isGiftUrlCreated && includeHighlights) {
+			return (
+				<div className="x-gift-article__checkbox">
+					<div className="x-gift-article__highlight-shared">highlights VISIBLE TO RECIPIENTS</div>
```
Comment (fabiosikansi) [inline components/x-gift-article/src/HighlightSection.jsx:15]: why there a difference of case here? the uppercase should be set via text-transformation

File: components/x-gift-article/src/GiftArticle.jsx [hunk 5]
```diff
@@ -188,6 +193,13 @@ const withGiftFormActions = withActions(
 						}
 					}
 				}
+			},
+			includeHighlightsHandler() {
+				return (state) => {
+					const includeHighlights = !state.includeHighlights
+					initialProps.includeHighlights = includeHighlights
+					return { includeHighlights }
+				}
```
Comment (fabiosikansi) [inline components/x-gift-article/src/GiftArticle.jsx:202]: why is this action changing a property in initialProps? this doesn't look like a good pattern to me, shouldn't it only update the state?

**Review by AnnaDraganova — APPROVED** [review]

**Comment by fabiosikansi** [comment]
This work was been merged in other PRs and later replaced by PR #713

**Pull request closed without merging**